recipe-server/client/control/components/MultiPicker.js

```diff
@@ -51,88 +44,65 @@ export default class MultiPicker extends React.Component {
       value,
     } = this.props;
 
-    const selectedOptions = options.filter(option =>
-      value.indexOf(option.value) === -1);
-
-    let displayedOptions = [].concat(selectedOptions)
-      .map(option => option.value)
-      .map(val =>
-        (!val ? null : { ...this.convertValueToObj(val) }))
-      .filter(x => x);
-
     const {
       filterText,
     } = this.state;
 
-    if (filterText) {
-      displayedOptions = displayedOptions.filter(option =>
-        JSON.stringify(option).indexOf(filterText) > -1);
-    }
+    let selectedOptions = options.filter(option => value.indexOf(option.value) === -1);
 
-    return displayedOptions;
-  }
-
-  handleApplyOption(event) {
-    event.persist();
-
-    if (!this.availableRef) {
-      return;
-    }
-
-    const {
-      value,
-      onChange,
-    } = this.props;
-
-    let selectedFilters = MultiPicker.getActiveSelectOptions(this.availableRef);
+    if (filterText) {
+      const searchText = filterText.toLowerCase();
 
-    if (!selectedFilters || selectedFilters.length === 0) {
-      selectedFilters = this.getDisplayedOptions();
-      selectedFilters = selectedFilters.map(option => option.value);
+      selectedOptions = selectedOptions.filter(option => {
+        const searchValues = JSON.stringify(Object.values(option)).toLowerCase();
```

Comment:
The `option`s only have two fields - `label` (displayed name) and `value` (the abbreviation), e.g. `{ label: 'Albania', value: 'AL'}`, so I don't think it's necessary to manually pluck those values out.

As for the JSON business, you're correct that searching with quotes/commas would produce unintended search results. How do you feel about:
```
selectedOptions = selectedOptions.filter(option => {
  const searchValues = Object.values(option).join(' ').toLowerCase();
  return searchValues.indexOf(searchText) > -1;
});
```
It still looks over only the values, but becomes a concatenated string without any extraneous quotes/commas.